### PR TITLE
Feat/sdk v2 settle flow

### PR DESCRIPTION
**Pull request by ioanaroceanu2000**

### CODE REVIEW

File: packages/sdk/src/services/editLp/editLp.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+import { getTokenDetails, scale, descale } from '@voltz-protocol/commons-v2';
+import { notionalToLiquidityBN } from '../../utils/helpers';
+import { defaultAbiCoder } from 'ethers/lib/utils';
+import { CompleteEditLpDetails, LpArgs, LpPeripheryParameters } from './types';
+import { getEditLpPeripheryParams } from './getEditLpPeripheryParams';
+import { InfoPostLp } from '../lp';
+import { encodeLp } from '../lp/encode';
+
+async function createLpParams({
+  poolId,
```
Comment (arturbeg) [inline packages/sdk/src/services/editLp/editLp.ts:20]: think if we have access to positionId, then we should be able to derive the poolId from the positionId (see v1 sdk editSwap as an example, wdyt?

File: packages/sdk/src/services/editLp/editLp.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    fixedRateLower,
+    fixedRateUpper,
+  });
+
+  const { data, value, chainId } = await getLpTxData(params);
+  const result = await executeTransaction(signer, data, value, chainId);
+  return result;
+}
+
+export async function getInfoPostLp({
```
Comment (arturbeg) [inline packages/sdk/src/services/editLp/editLp.ts:75]: worth having one getInfoPostLp and one for getInfoPostLpEdit? one requires the poolId, the other requires the positionId to align with v1 unless you think there's a good reason to go with this pattern?

File: packages/sdk/src/utils/math/tickHelpers.ts [hunk 3]
```diff
@@ -139,3 +140,12 @@ function tickToFixedRate(tick: number) {
 
   return priceToFixedRate(price);
 }
+
+export function fixedRateToPrice(fixedRate: number): string {
```
Comment (arturbeg) [inline packages/sdk/src/utils/math/tickHelpers.ts:144]: wonder if some of this tick math that's shared across v1 and v2 vamms can be abstracted away into commons type of package?

File: packages/sdk/src/utils/helpers.ts [hunk 4]
```diff
@@ -1,24 +1,38 @@
...
-  priceLower?: number,
-  priceUpper?: number,
-): Promise<string> {
+export async function createAccountId({
+  ownerAddress,
+  productAddress,
+  marketId,
+  maturityTimestamp,
+  isLp,
+  priceLower,
```
Comment (arturbeg) [inline packages/sdk/src/utils/helpers.ts:25]: do accountIds of takers also need priceLower and priceUpper or is this just a requirement for makers ?

File: packages/sdk/src/services/editSwap/editSwap.ts [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+import {
+  baseAmountToNotionalBN,
+  notionalToBaseAmount,
+} from '../../utils/helpers';
+import { VERY_BIG_NUMBER, ZERO_BN } from '../../utils/constants';
+import { fixedRateToPrice } from '../../utils/math/tickHelpers';
+import { decodeSwapOutput, InfoPostSwap, processInfoPostSwap } from '../swap';
+
+async function createEditSwapParams({
+  poolId,
```
Comment (arturbeg) [inline packages/sdk/src/services/editSwap/editSwap.ts:24]: can we avoid the need for poolId in context of edit swap and derive it from the position (see v1-sdk editSwap)

**Review by arturbeg — APPROVED** [review]

**Comment by arturbeg** [comment]
@ssbarbee feel free to jump in ser, sdk v2 integration is cooking:)

**Comment by ssbarbee** [comment]
> @ssbarbee feel free to jump in with a review ser, sdk v2 integration is cooking:)

@arturbeg as long as @ioanaroceanu2000 follows the v1-stateless SDK integration we should be fine. Just follow same params/response and we gucci

File: packages/sdk/src/services/editLp/editLp.ts (hunk 1 above)
Comment (ioanaroceanu2000) [reply packages/sdk/src/services/editLp/editLp.ts]: true, I can just have position Id

File: packages/sdk/src/utils/helpers.ts (hunk 4 above)
Comment (ioanaroceanu2000) [reply packages/sdk/src/utils/helpers.ts:25]: just for makers

**Pull request merged**